dockerfiles/Dockerfile.dev

```diff
@@ -1,33 +1,49 @@
+# syntax=docker/dockerfile:1.1.7-experimental
 ARG GO_VERSION=1.13.15
 
-FROM    golang:${GO_VERSION}-alpine
-
-RUN     apk add --no-cache bash build-base ca-certificates coreutils curl git
-
-# v0.1.1
-ARG     VNDR_SHA=85886e1ac99b8d96590e6e0d9f075dc7a711d132
-RUN     go get -d github.com/LK4D4/vndr && \
-        cd /go/src/github.com/LK4D4/vndr && \
-        git checkout -q "$VNDR_SHA" && \
-        go build -v -o /usr/bin/vndr . && \
-        rm -rf /go/src/* /go/pkg/* /go/bin/*
-
-# v0.2.0
-ARG     ESC_SHA=0ea7db170df78dcddf3e223365f444163147fe89
-RUN     go get -d github.com/mjibson/esc && \
-        cd /go/src/github.com/mjibson/esc && \
-        git checkout -q "$ESC_SHA" && \
-        go build -v -o /usr/bin/esc . && \
-        rm -rf /go/src/* /go/pkg/* /go/bin/*
-
-ARG     GOTESTSUM_VERSION=0.4.0
-RUN     curl -Ls https://github.com/gotestyourself/gotestsum/releases/download/v${GOTESTSUM_VERSION}/gotestsum_${GOTESTSUM_VERSION}_linux_amd64.tar.gz -o gotestsum.tar.gz && \
-        tar -xf gotestsum.tar.gz gotestsum -C /usr/bin && \
-        rm gotestsum.tar.gz
-
-ENV     CGO_ENABLED=0 \
-        PATH=$PATH:/go/src/github.com/docker/cli/build \
-        DISABLE_WARN_OUTSIDE_CONTAINER=1
+FROM golang:${GO_VERSION}-alpine AS golang
+ENV  CGO_ENABLED=0
+
+FROM golang AS esc
+ARG GO111MODULE=on
```

Comment:
What about adding another stage which adds `GO111MODULE=on` as `ENV` instead?
```dockerfile
FROM golang AS golang-build
ENV GO111MODULE=on

FROM golang-build AS esc
...

FROM golang AS dev
```

Reply:
I used `ARG` so that the `GO111MODULE` wasn't persisted, and the final stage currently doesn't use Go modules.

Per our discussion on slack, I changed that to jus inline the `GO111MODULE=on` instead of using an `ARG`